### ISSUE TITLE
MessageNotUnderstood: LBGitFileListDiff>>asDiffTree

### ISSUE THREAD

**Issue by MrModder** [LeonBein/SmalltalkGitTesting#21]
```
13 December 2018 5:37:55.252797 pm

VM: Win32 - Smalltalk
Image: Squeak5.2 [latest update: #18228]

SecurityManager state:
Restricted: false
FileAccess: true
SocketAccess: true
Working Dir C:\Users\Leon\Documents\Leons Dokumente\HPI\18_WiSe\BP\Squeak\images
Trusted Dir C:\Users\Leon\Documents\Leons Dokumente\HPI\18_WiSe\BP\Squeak\images\Leon
Untrusted Dir C:\Users\Leon\Documents\My Squeak

LBGitFileListDiff(Object)>>doesNotUnderstand: #asDiffTree
	Receiver: a LBGitFileListDiff
	Arguments and temporary variables: 
		aMessage: 	asDiffTree
		exception: 	MessageNotUnderstood: LBGitFileListDiff>>asDiffTree
		resumeValue: 	nil
	Receiver's instance variables: 
		left: 	a LBGitRecordedFilesArtifact(On commit: 'Add configuration-prototype')
		right: 	a LBGitLiveFilesArtifact(live)
		added: 	#()
		removed: 	#()
		stayed: 	#('travis-traffic-light/.gitignore' 'travis-traffic-light/StartTravisTr...etc...
		modified: 	#('travis-traffic-light/package-lock.json' 'travis-traffic-light/trav...etc...
		diffs: 	an OrderedCollection(a LBGitFileModification a LBGitFileModification a LBGitFileModification...etc...
		target: 	nil

[] in SquotSaveVersionDialog>>rootNodes
	Receiver: a SquotSaveVersionDialog
	Arguments and temporary variables: 
		each: 	a LBGitFileListDiff
	Receiver's instance variables: 
		dependents: 	a DependentsArray(a PluggableSystemWindow<Untitled Window>(1005003)...etc...
		message: 	'Enter commit log message here'
		canceled: 	nil
		diffRootNodes: 	an OrderedCollection()
		selectedDiffNode: 	nil
		controllerForIgnores: 	BP
		editedPatch: 	a SquotPatch

[] in Dictionary>>valuesDo:
	Receiver: a Dictionary(''->a LBGitFileListDiff )
	Arguments and temporary variables: 
		aBlock: 	''->a LBGitFileListDiff
		association: 	[closure] in SquotSaveVersionDialog>>rootNodes
	Receiver's instance variables: 
		tally: 	1
		array: 	{nil . nil . ''->a LBGitFileListDiff . nil . nil}

Dictionary>>associationsDo:
	Receiver: a Dictionary(''->a LBGitFileListDiff )
	Arguments and temporary variables: 
		aBlock: 	[closure] in Dictionary>>valuesDo:
		element: 	''->a LBGitFileListDiff
		index: 	3
		indexLimiT: 	5
	Receiver's instance variables: 
		tally: 	1
		array: 	{nil . nil . ''->a LBGitFileListDiff . nil . nil}

Dictionary>>valuesDo:
	Receiver: a Dictionary(''->a LBGitFileListDiff )
	Arguments and temporary variables: 
		aBlock: 	[closure] in SquotSaveVersionDialog>>rootNodes
	Receiver's instance variables: 
		tally: 	1
		array: 	{nil . nil . ''->a LBGitFileListDiff . nil . nil}

Dictionary>>do:
	Receiver: a Dictionary(''->a LBGitFileListDiff )
	Arguments and temporary variables: 
		aBlock: 	[closure] in SquotSaveVersionDialog>>rootNodes
	Receiver's instance variables: 
		tally: 	1
		array: 	{nil . nil . ''->a LBGitFileListDiff . nil . nil}

SquotPatch>>diffsDo:
	Receiver: a SquotPatch
	Arguments and temporary variables: 
		aBlock: 	[closure] in SquotSaveVersionDialog>>rootNodes
	Receiver's instance variables: 
		diffs: 	a Dictionary(''->a LBGitFileListDiff )
		loadOrderDiff: 	a DiffPatch

SquotSaveVersionDialog>>rootNodes
	Receiver: a SquotSaveVersionDialog
	Arguments and temporary variables: 

	Receiver's instance variables: 
		dependents: 	a DependentsArray(a PluggableSystemWindow<Untitled Window>(1005003)...etc...
		message: 	'Enter commit log message here'
		canceled: 	nil
		diffRootNodes: 	an OrderedCollection()
		selectedDiffNode: 	nil
		controllerForIgnores: 	BP
		editedPatch: 	a SquotPatch

PluggableTreeMorph>>update:
	Receiver: a PluggableTreeMorph<hierarchical diff>(1572336)
	Arguments and temporary variables: 
		what: 	#rootNodes
	Receiver's instance variables: 
		bounds: 	0@0 corner: 150@120
		owner: 	nil
		submorphs: 	{a TransformMorph(1046128)}
		fullBounds: 	nil
		color: 	Color tungsten
		extension: 	a MorphExtension (3953635) [externalName = hierarchical diff ]  [eve...etc...
		borderWidth: 	1
		borderColor: 	Color iron
		model: 	a SquotSaveVersionDialog
		slotName: 	nil
		open: 	false
		scrollBar: 	a ScrollBar(3769387)
		scroller: 	a TransformMorph(1046128)
		retractableScrollBar: 	false
		scrollBarOnLeft: 	false
		getMenuSelector: 	#diffTreeMenu:
		getMenuTitleSelector: 	nil
		hasFocus: 	false
		hScrollBar: 	a ScrollBar(1328096)
		hScrollBarPolicy: 	#never
		vScrollBarPolicy: 	#whenNeeded
		scrollBarThickness: 	10
		selectedMorph: 	nil
		hoveredMorph: 	nil
		getListSelector: 	nil
		keystrokeActionSelector: 	nil
		autoDeselect: 	nil
		columns: 	nil
		columnsCache: 	nil
		sortingSelector: 	nil
		getSelectionSelector: 	#diffSelectedNode
		setSelectionSelector: 	#selectNode:
		potentialDropMorph: 	nil
		lineColor: 	(Color r: 0.659 g: 0.659 b: 0.659)
		font: 	a StrikeFont(Darkmap DejaVu Sans9 14)
		textColor: 	(Color r: 0.941 g: 0.941 b: 0.941)
		rootWrappers: 	nil
		selectedWrapper: 	nil
		getRootsSelector: 	#rootNodes
		getChildrenSelector: 	#diffChildrenOf:
		hasChildrenSelector: 	#diffHasChildren:
		getLabelSelector: 	#nodeLabelOf:
		getIconSelector: 	nil
		getSelectedPathSelector: 	#diffSelectedPath
		setSelectedPathSelector: 	nil
		setSelectedParentSelector: 	nil
		getHelpSelector: 	nil
		dropItemSelector: 	nil
		wantsDropSelector: 	nil
		dragItemSelector: 	nil
		dragTypeSelector: 	nil
		nodeClass: 	SquotDiffNodeTreeWrapper
		lastKeystrokeTime: 	0
		lastKeystrokes: 	''
		dragStartedSelector: 	nil
		doubleClickSelector: 	nil

PluggableTreeMorph>>getRootsSelector:
	Receiver: a PluggableTreeMorph<hierarchical diff>(1572336)
	Arguments and temporary variables: 
		aSelector: 	#rootNodes
	Receiver's instance variables: 
		bounds: 	0@0 corner: 150@120
		owner: 	nil
		submorphs: 	{a TransformMorph(1046128)}
		fullBounds: 	nil
		color: 	Color tungsten
		extension: 	a MorphExtension (3953635) [externalName = hierarchical diff ]  [eve...etc...
		borderWidth: 	1
		borderColor: 	Color iron
		model: 	a SquotSaveVersionDialog
		slotName: 	nil
		open: 	false
		scrollBar: 	a ScrollBar(3769387)
		scroller: 	a TransformMorph(1046128)
		retractableScrollBar: 	false
		scrollBarOnLeft: 	false
		getMenuSelector: 	#diffTreeMenu:
		getMenuTitleSelector: 	nil
		hasFocus: 	false
		hScrollBar: 	a ScrollBar(1328096)
		hScrollBarPolicy: 	#never
		vScrollBarPolicy: 	#whenNeeded
		scrollBarThickness: 	10
		selectedMorph: 	nil
		hoveredMorph: 	nil
		getListSelector: 	nil
		keystrokeActionSelector: 	nil
		autoDeselect: 	nil
		columns: 	nil
		columnsCache: 	nil
		sortingSelector: 	nil
		getSelectionSelector: 	#diffSelectedNode
		setSelectionSelector: 	#selectNode:
		potentialDropMorph: 	nil
		lineColor: 	(Color r: 0.659 g: 0.659 b: 0.659)
		font: 	a StrikeFont(Darkmap DejaVu Sans9 14)
		textColor: 	(Color r: 0.941 g: 0.941 b: 0.941)
		rootWrappers: 	nil
		selectedWrapper: 	nil
		getRootsSelector: 	#rootNodes
		getChildrenSelector: 	#diffChildrenOf:
		hasChildrenSelector: 	#diffHasChildren:
		getLabelSelector: 	#nodeLabelOf:
		getIconSelector: 	nil
		getSelectedPathSelector: 	#diffSelectedPath
		setSelectedPathSelector: 	nil
		setSelectedParentSelector: 	nil
		getHelpSelector: 	nil
		dropItemSelector: 	nil
		wantsDropSelector: 	nil
		dragItemSelector: 	nil
		dragTypeSelector: 	nil
		nodeClass: 	SquotDiffNodeTreeWrapper
		lastKeystrokeTime: 	0
		lastKeystrokes: 	''
		dragStartedSelector: 	nil
		doubleClickSelector: 	nil

MorphicToolBuilder>>buildPluggableTree:
	Receiver: a MorphicToolBuilder
	Arguments and temporary variables: 
		aSpec: 	a PluggableTreeSpec
		widget: 	a PluggableTreeMorph<hierarchical diff>(1572336)
		policy: 	nil
	Receiver's instance variables: 
		parent: 	a PluggableSystemWindow<Untitled Window>(1005003)
		widgets: 	a Dictionary('hierarchical diff'->a PluggableTreeMorph<hierarchical di...etc...
		panes: 	an OrderedCollection()
		parentMenu: 	nil

PluggableTreeSpec>>buildWith:
	Receiver: a PluggableTreeSpec
	Arguments and temporary variables: 
		builder: 	a MorphicToolBuilder
	Receiver's instance variables: 
		name: 	'hierarchical diff'
		help: 	nil
		model: 	a SquotSaveVersionDialog
		frame: 	0@0 corner: 0.4@0.7
		color: 	nil
		minimumExtent: 	nil
		margin: 	nil
		padding: 	nil
		horizontalResizing: 	nil
		verticalResizing: 	nil
		roots: 	#rootNodes
		getSelectedPath: 	#diffSelectedPath
		setSelectedPath: 	nil
		setSelected: 	#selectNode:
		getSelected: 	#diffSelectedNode
		setSelectedParent: 	nil
		getChildren: 	#diffChildrenOf:
		hasChildren: 	#diffHasChildren:
		label: 	#nodeLabelOf:
		icon: 	nil
		unusedVar: 	nil
		menu: 	#diffTreeMenu:
		keyPress: 	nil
		doubleClick: 	nil
		dropItem: 	nil
		dropAccept: 	nil
		autoDeselect: 	nil
		dragItem: 	nil
		nodeClass: 	SquotDiffNodeTreeWrapper
		columns: 	nil
		vScrollBarPolicy: 	nil
		hScrollBarPolicy: 	nil
		dragStarted: 	nil

[] in MorphicToolBuilder(ToolBuilder)>>buildAll:in:
	Receiver: a MorphicToolBuilder
	Arguments and temporary variables: 
<<error during printing>
	Receiver's instance variables: 
		parent: 	a PluggableSystemWindow<Untitled Window>(1005003)
		widgets: 	a Dictionary('hierarchical diff'->a PluggableTreeMorph<hierarchical di...etc...
		panes: 	an OrderedCollection()
		parentMenu: 	nil

Array(SequenceableCollection)>>do:
	Receiver: {a PluggableTreeSpec . a PluggableTextSpec . a PluggableTextSpec . a PluggablePanelSpec}
	Arguments and temporary variables: 
		aBlock: 	[closure] in MorphicToolBuilder(ToolBuilder)>>buildAll:in:
		index: 	1
		indexLimiT: 	4
	Receiver's instance variables: 
{a PluggableTreeSpec . a PluggableTextSpec . a PluggableTextSpec . a PluggablePanelSpec}

MorphicToolBuilder(ToolBuilder)>>buildAll:in:
	Receiver: a MorphicToolBuilder
	Arguments and temporary variables: 
		aList: 	{a PluggableTreeSpec . a PluggableTextSpec . a PluggableTextSpec . a PluggablePanelSpec...etc...
		newParent: 	a PluggableSystemWindow<Untitled Window>(1005003)
		prior: 	nil
	Receiver's instance variables: 
		parent: 	a PluggableSystemWindow<Untitled Window>(1005003)
		widgets: 	a Dictionary('hierarchical diff'->a PluggableTreeMorph<hierarchical di...etc...
		panes: 	an OrderedCollection()
		parentMenu: 	nil

MorphicToolBuilder>>buildPluggableWindow:
	Receiver: a MorphicToolBuilder
	Arguments and temporary variables: 
		aSpec: 	a PluggableWindowSpec
		widget: 	a PluggableSystemWindow<Untitled Window>(1005003)
		label: 	nil
	Receiver's instance variables: 
		parent: 	a PluggableSystemWindow<Untitled Window>(1005003)
		widgets: 	a Dictionary('hierarchical diff'->a PluggableTreeMorph<hierarchical di...etc...
		panes: 	an OrderedCollection()
		parentMenu: 	nil

PluggableWindowSpec>>buildWith:
	Receiver: a PluggableWindowSpec
	Arguments and temporary variables: 
		builder: 	a MorphicToolBuilder
	Receiver's instance variables: 
		name: 	nil
		help: 	nil
		model: 	a SquotSaveVersionDialog
		frame: 	nil
		color: 	nil
		minimumExtent: 	nil
		margin: 	nil
		padding: 	nil
		horizontalResizing: 	nil
		verticalResizing: 	nil
		children: 	{a PluggableTreeSpec . a PluggableTextSpec . a PluggableTextSpec . a PluggablePanelSpec...etc...
		layout: 	nil
		wantsResizeHandles: 	nil
		spacing: 	nil
		label: 	'New version'
		extent: 	nil
		closeAction: 	nil
		multiWindowStyle: 	nil

MorphicToolBuilder(ToolBuilder)>>build:
	Receiver: a MorphicToolBuilder
	Arguments and temporary variables: 
		anObject: 	a PluggableWindowSpec
	Receiver's instance variables: 
		parent: 	a PluggableSystemWindow<Untitled Window>(1005003)
		widgets: 	a Dictionary('hierarchical diff'->a PluggableTreeMorph<hierarchical di...etc...
		panes: 	an OrderedCollection()
		parentMenu: 	nil

SquotSaveVersionDialog>>buildWith:
	Receiver: a SquotSaveVersionDialog
	Arguments and temporary variables: 
		builder: 	a MorphicToolBuilder
		windowWidgets: 	{a PluggableTreeSpec . a PluggableTextSpec . a PluggableTextSpec...etc...
		window: 	a PluggableWindowSpec
	Receiver's instance variables: 
		dependents: 	a DependentsArray(a PluggableSystemWindow<Untitled Window>(1005003)...etc...
		message: 	'Enter commit log message here'
		canceled: 	nil
		diffRootNodes: 	an OrderedCollection()
		selectedDiffNode: 	nil
		controllerForIgnores: 	BP
		editedPatch: 	a SquotPatch


--- The full stack ---
LBGitFileListDiff(Object)>>doesNotUnderstand: #asDiffTree
[] in SquotSaveVersionDialog>>rootNodes
[] in Dictionary>>valuesDo:
Dictionary>>associationsDo:
Dictionary>>valuesDo:
Dictionary>>do:
SquotPatch>>diffsDo:
SquotSaveVersionDialog>>rootNodes
PluggableTreeMorph>>update:
PluggableTreeMorph>>getRootsSelector:
MorphicToolBuilder>>buildPluggableTree:
PluggableTreeSpec>>buildWith:
[] in MorphicToolBuilder(ToolBuilder)>>buildAll:in:
Array(SequenceableCollection)>>do:
MorphicToolBuilder(ToolBuilder)>>buildAll:in:
MorphicToolBuilder>>buildPluggableWindow:
PluggableWindowSpec>>buildWith:
MorphicToolBuilder(ToolBuilder)>>build:
SquotSaveVersionDialog>>buildWith:
 - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
MorphicToolBuilder(ToolBuilder)>>build:
MorphicToolBuilder>>open:
MorphicToolBuilder>>open:label:
SquotSaveVersionDialog>>requestSelectedChangesWithTitle:
SquotWorkingCopy>>selectFromPatch:forSaveWithTitle:ifCanceled:
SquotWorkingCopy>>saveNewVersionFromPatch:ifCanceled:
SquotWorkingCopy>>saveNewVersionInteractivelyIfCanceled:
SquitBrowser>>actionCommit
PluggableButtonMorphPlus(PluggableButtonMorph)>>performAction
PluggableButtonMorphPlus>>performAction
[] in PluggableButtonMorphPlus(PluggableButtonMorph)>>mouseUp:
Array(SequenceableCollection)>>do:
PluggableButtonMorphPlus(PluggableButtonMorph)>>mouseUp:
PluggableButtonMorphPlus(Morph)>>handleMouseUp:
MouseButtonEvent>>sentTo:
PluggableButtonMorphPlus(Morph)>>handleEvent:
PluggableButtonMorphPlus(Morph)>>handleFocusEvent:
MorphicEventDispatcher>>doHandlingForFocusEvent:with:
MorphicEventDispatcher>>dispatchFocusEvent:with:
PluggableButtonMorphPlus(Morph)>>processFocusEvent:using:
PluggableButtonMorphPlus(Morph)>>processFocusEvent:
[] in [] in [] in HandMorph>>sendFocusEvent:to:clear:
BlockClosure>>ensure:
MouseButtonEvent(MorphicEvent)>>becomeActiveDuring:
[] in [] in HandMorph>>sendFocusEvent:to:clear:
BlockClosure>>ensure:
HandMorph>>becomeActiveDuring:
[] in HandMorph>>sendFocusEvent:to:clear:
BlockClosure>>ensure:
PasteUpMorph>>becomeActiveDuring:
HandMorph>>sendFocusEvent:to:clear:
HandMorph>>sendEvent:focus:clear:
HandMorph>>sendMouseEvent:
HandMorph>>handleEvent:
HandMorph>>processEvents
[] in AnimWorldState(WorldState)>>doOneCycleNowFor:
Array(SequenceableCollection)>>do:
AnimWorldState(WorldState)>>handsDo:
AnimWorldState(WorldState)>>doOneCycleNowFor:
AnimWorldState(WorldState)>>doOneCycleFor:
PasteUpMorph>>doOneCycle
[] in AnimMorphicProject>>spawnNewProcess
-- and more not shown --```

**Comment by LeonBein** [comment]
Try this:
asDiffTree

	^ SquotDiffNode 
		title: '' 
		content: self 
		children: self topNodes

**Comment by LeonBein** [comment]
Fixed by 7792999fdc475d29c8fa961d867908815c004042
